### PR TITLE
Feature/Sign up google

### PR DESCRIPTION
**Pull request by jphinning**
Hi Luda,

In this PR I've created the google OAuth component button that is being used on Sign In and Sign Up pages:

image

Also used RTK query to make a request to the /auth/google/success route to get the JWT token from the backend after authenticating with Google. 

Finally, I've made sure that after a local or google login, the user persisted state is updated with the help of the useAuth() hook.

### CODE REVIEW

File: src/components/UI/googleAuthButton/GoogleAuthButton.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  children: React.ReactNode;
+}
+
+function GoogleAuthButton({ children }: GoogleAuthButtonProps) {
+  const googleLogin = () => {
+    window.open(`${process.env.REACT_APP_API_URL}/auth/google`, "_self");
+  };
+
+  return (
+    <GoogleButton onClick={() => googleLogin()}>
```
Comment (lyudad) [inline src/components/UI/googleAuthButton/GoogleAuthButton.tsx]: onClick={googleLogin}>

File: src/hooks/useAuth.tsx [hunk 2]
```diff
@@ -26,14 +29,15 @@ const useAuth = () => {
       setSignOut({
         authToken: undefined,
         email: undefined,
+        id: undefined,
         role: undefined,
       })
```
Comment (lyudad) [inline src/hooks/useAuth.tsx]: can we make setSignOut(null)? instead describe each field?

File: src/components/UI/googleAuthButton/GoogleAuthButton.tsx (hunk 1 above)
Comment (jphinning) [reply src/components/UI/googleAuthButton/GoogleAuthButton.tsx]: Done :thumbsup:

File: src/hooks/useAuth.tsx (hunk 2 above)
Comment (jphinning) [reply src/hooks/useAuth.tsx]: Modified the reducer setSignOut to return to the initial state. Now I can call setSignOut() without any object parameters. 

```typescript
setSignOut: () => initialState,
```

**Pull request merged**